src/domain/core/src/utils/owned_file.rs

```diff
@@ -14,7 +14,7 @@ use std::path::{Path, PathBuf};
 /// Represents a file on local FS that should either be consumed using
 /// [`OwnedFile::into_inner`()] or will be cleaned up on drop. Is typically used
 /// to commit data and checkpoints generated by the engines into a dataset.
-#[derive(Debug)]
+#[derive(Debug, Clone)]
```

Comment:
Oh no - this is super dangerous!

This struct is meant to represent ownerwhip over a file. Notice that it implements `Drop` that will delete the file if it was not consumed by `into_inner()`.

So if you make this `Clone` all hell breaks loose - it's just like if `Box<T>` was clonable.

Reply:
Yeah, it was stupid to add it. FIxed